### PR TITLE
Add new tunnels to enable ERC721 transfer with token URIs

### PR DESCRIPTION
**Pull request by QEDK**
* Add `FxERC721URIStorage` token type with corresponding interface
* Add child and root tunnels for `FxERC721URIStorage` tokens
* Add Etherscan plugin to verify contracts

### CODE REVIEW

File: contracts/lib/ERC721URIStorage.sol [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./ERC721.sol";
+
+/**
+ * @dev ERC721 token with storage based token URI management.
+ */
+abstract contract ERC721URIStorage is ERC721 {
```
Comment (imyourm8) [inline contracts/lib/ERC721URIStorage.sol:10]: Better name it as ERC721WithURI or something like that. "Storage" misleads the end-user, also it is used in proxies.

File: contracts/tokens/IFxERC721URIStorage.sol [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+interface IFxERC721URIStorage {
```
Comment (imyourm8) [inline contracts/tokens/IFxERC721URIStorage.sol:4]: Same here, naming

File: contracts/tokens/FxERC721URIStorage.sol [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+pragma solidity ^0.8.0;
+
+import { ERC721URIStorage } from "../lib/ERC721URIStorage.sol";
+import { IFxERC721URIStorage } from "./IFxERC721URIStorage.sol";
+
+
+/**
+ * @title FxERC20 represents fx erc20
+ */
+contract FxERC721URIStorage is IFxERC721URIStorage, ERC721URIStorage {
```
Comment (imyourm8) [inline contracts/tokens/FxERC721URIStorage.sol:11]: And here, naming

**Review by imyourm8 — CHANGES_REQUESTED** [review]

**Review by QEDK — COMMENTED** [review]
I can't seem to delete this so I'll just leave this here I guess...

File: contracts/lib/ERC721URIStorage.sol (hunk 1 above)
Comment (QEDK) [reply contracts/lib/ERC721URIStorage.sol:10]: @imyourm8 this is intentional, see https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/ERC721URIStorage.sol which is the most common implementation of ERC721 with URI (and this contract is mostly the OZ implementation) No problems with renaming the tunnels though, let me know.

Comment (imyourm8) [reply contracts/lib/ERC721URIStorage.sol:10]: Okay

**Pull request closed without merging**